### DISCUSSION TITLE
Creating a custom vega-template dvc with tool-tip images

### DISCUSSION THREAD

**Discussion by ayman-saleh** [treeverse/dvc#8552] [Help]
Hello,

I created a vega plot that is a scatter plot with a tool-tip when you hover over a data point to look at an image.

I want to embed this vega plot into dvc, but it seems that I have to create a dvc template.

In order to render the base64 encoded image when you hover over a datapoint, you need to read data with the key `image` otherwise it will not render correctly.

How do i reference a created template within `dvc.yaml` under plots?

**Comment by pmrowla** [comment]
If you are using the top level `plots` field in your `dvc.yaml`, you would specify the custom template in
```
plots:
  - <data series source>
    template: <template name>
    ...
```
(see the example in the docs: https://dvc.org/doc/user-guide/experiment-management/visualizing-plots#visualizing-plots)

For templates stored in `.dvc/plots` (default location for custom templates), the path and the json extension are not required. Otherwise, you would specify path + file extension to your template (where the path is relative to the `dvc.yaml` location)
